src/app/courses/step-view-courses/courses-step-view.component.ts

```diff
@@ -33,15 +36,36 @@ export class CoursesStepViewComponent implements OnInit, OnDestroy {
     private route: ActivatedRoute,
     private coursesService: CoursesService,
     private userService: UserService,
-    private submissionsService: SubmissionsService
+    private submissionsService: SubmissionsService,
+    private resourcesService: ResourcesService
   ) {}
 
   ngOnInit() {
-    this.coursesService.courseUpdated$
-    .pipe(takeUntil(this.onDestroy$))
+    this.coursesService.courseUpdated$.pipe(takeUntil(this.onDestroy$))
     .subscribe(({ course, progress = [] }: { course: any, progress: any }) => {
-      this.initCourse(course, progress);
+      this.course = course;
```

Comment:
You don't need to store the whole course on the component.  The resources array is found on `stepDetails`.